### PR TITLE
Added maxmemory option support for virt-install

### PR DESCRIPTION
**Pull request by sathnaga**
Added mammemory option support for virt-install
Added a method to check if a sub-option present in the
virt-install command line for a given parameter

### CODE REVIEW

**Comment by sathnaga** [comment]
@will-Do could you please check and merge if fine, Thanks in advance :-)

File: virttest/libvirt_vm.py [hunk 1]
```diff
@@ -370,6 +370,12 @@ def has_option(help_text, option):
         def has_os_variant(os_text, os_variant):
             return bool(re.search(r"%s" % os_variant, os_text, re.MULTILINE))
 
+        def has_sub_option(option, sub_option):
+            option_help_text = process.system_output("%s --%s help" %
+                                                     (virt_install_binary, option),
+                                                     verbose=False)
+            return bool(re.search(r"%s" % sub_option, option_help_text, re.MULTILINE))
```
Comment (will-Do) [inline virttest/libvirt_vm.py]: It's not the perfect place to add this function here, but if you can't find a more appropriate place, lets keep it here

File: virttest/libvirt_vm.py [hunk 2]
```diff
@@ -394,8 +400,15 @@ def add_hvm_or_pv(help_text, hvm_or_pv):
                 logging.warning("Unknown virt type hvm_or_pv, using default.")
                 return ""
 
-        def add_mem(help_text, mem):
-            return " --ram=%s" % mem
+        def add_mem(help_text, mem, maxmem=None):
+            logging.debug("satheesh")
```
Comment (will-Do) [inline virttest/libvirt_vm.py]: This should be removed

**Comment by will-Do** [comment]
And there's a typo in your commit message
s/mammemory/maxmemory/

File: virttest/libvirt_vm.py (hunk 1 above)
Comment (sathnaga) [inline virttest/libvirt_vm.py]: @will-Do Other helper functions for command line sits there, that's the reason have placed it,
not finding any other appropriate place right now, Hope it should be fine?

File: virttest/libvirt_vm.py (hunk 2 above)
Comment (sathnaga) [inline virttest/libvirt_vm.py]: @will-Do Sorry, typo, removed

**Comment by sathnaga** [comment]
@will-Do Tnx, have fixed the comments and pushed :-)

**Comment by sathnaga** [comment]
@will-Do Could you please merge this PR, thanks in advance :-)

**Comment by sathnaga** [comment]
@will-Do Thanks :-)

**Pull request merged**